include/multipass/utils.h

```diff
@@ -127,10 +129,69 @@ std::string qenum_to_string(RegisteredQtEnum val);
 // other helpers
 QString get_multipass_storage();
 QString make_uuid(const std::optional<std::string>& seed = std::nullopt);
+
 template <typename OnTimeoutCallable, typename TryAction, typename... Args>
 void try_action_for(OnTimeoutCallable&& on_timeout, std::chrono::milliseconds timeout, TryAction&& try_action,
                     Args&&... args);
 
+template <typename T>
+bool is_default_constructed(const T& input_type)
+{
+    return input_type == T{};
+}
+
+// simplified parallel transform, it takes a std container and a unary operation and
+// returns a std::vector<OutputValueType> where the OutputValueType is the unary operation return type
+// There are two options of the return types, one is the one below and the other one is auto. Eventually, I went with
+// the std::invoke_result_t based one because it makes the function signature more expressive despite the fact that it
+// makes std::invoke_result_t<std::decay_t<UnaryOperation>, InputValueType> code duplicate.
+template <typename Container, typename UnaryOperation>
+std::vector<std::invoke_result_t<std::decay_t<UnaryOperation>, typename Container::value_type>>
+parallel_transform(const Container& input_container, UnaryOperation&& unary_op)
+{
+    using InputValueType = typename Container::value_type;
```

Comment:
I don't see `InputValueType` used anywhere else accept for the following `using OutputValueType = ...`.  I'm not sure there is any value in defining this.

Reply:
@georgeliao :point_up:

Reply:
Yep, this can be fixed. Originally, I thought `InputValueType` is more expressive and symmetric to `OutputValueType`. So It adds a small value even though it is only used once. However, putting `Container::value_type` in one line with `OutputValueType` might make things cleaner.

Reply:
Ok, I think it's fine the way it is.